micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -612,8 +620,15 @@ private Id getMappedId(Id id) {
 
     private Meter getOrCreateMeter(@Nullable DistributionStatisticConfig config,
             BiFunction<Id, /* Nullable Generic */ DistributionStatisticConfig, ? extends Meter> builder, Id originalId,
-            Id mappedId, Function<Meter.Id, ? extends Meter> noopBuilder) {
-        Meter m = meterMap.get(mappedId);
+            Function<Meter.Id, ? extends Meter> noopBuilder) {
+
+        Meter m = preFilterIdToMeterMap.get(originalId);
+        if (m != null && !isStaleId(originalId)) {
+            return m;
+        }
+
+        Id mappedId = getMappedId(originalId);
+        m = meterMap.get(mappedId);
```

Comment:
```suggestion
        m = meterMap.get(mappedId);
        
        if(m != null) {
            // If the mapping exists and the meter is marked stale, then this meter is no longer stale.
            synchronized (meterMapLock) {
                if(isStaleId(originalId)) {
                    unmarkStaleId(originalId);
                }
            }
            return m;
        }
```

Reply:
You're right. This issue would be unfortunate and somewhat common to run into, I suspect. I've committed a fix in c63b93a5ab9678e7ffe7053c08a6b9f843364542